Add simple backwards compatibility check for torch.package

PaliC opened the pull request:
Fixes #65154, tests for backwards compatibility of torch.package by checking if packages that were created before can still be loaded.

suo reviewed (changes requested):
Looks good overall! Just a few minor comments. Can you also update the PR title and description? Since those get committed, it's good to keep context for release notes, other teams, etc.

PaliC committed:
Add simple backwards compatibility check for torch.package (#66739)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/65154, tests for backwards compatibility of torch.package by checking if packages that were created before can still be loaded.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/66739

Reviewed By: suo

Differential Revision: D31771526

Pulled By: PaliC

fbshipit-source-id: 13e53c2d5986d0f17a6c967874e751430b6eb3f8
